Changed output to html

ms130 opened the pull request:
What?

- Changed Rmarkdown output to html
- Added floating contents list
- Removed manual section numbering. Replaced with automatic section numbering.

chloepugh reviewed (dismissed):
Looks good! Might need to delete the Word doc at some point, otherwise it won't get updated and might confuse participants if they download the wrong file.

Could also be worth considering outputting a md file as well as html, so that the md can be viewed on github without needing to download anything. Not critical for this PR though.

ms130 commented:
Thanks for the suggestions Chloe! I'll merge it into `main` after the course, so as not to confused anyone.

georginaeaton commented:
Hi @ms130 I noticed a couple of things in the word doc that I assume might need changing to the Rmd so you might as well update in this pull request if possible. 

When I looked, the word document doesn't currently include renv::restore() as it does in the participant code - could this be added please if it isn't currently in the Rmd file (with a small bit of explanation around it). Also, Q3 in Exercises 2.7 should be either rewritten or should include a comma - it's not easy to read at the moment. I suggest changing it to something like the following: 3. Changing the SENTENCE class to factor. Then output the levels of this variable.

ms130 commented:
Main new changes made to Intro_markdown.Rmd:
- Added a point to "Basic setup" to install the `renv` package
- Updated §2.1 working to emphasis RStudio projects, that automatically set the working directory
- §2.3.2 used tidyverse `read_csv`
- Changed §2.7 Q3 wording
- Added the `count` function to §3.2 and 3.3
- §4.1 replace weekday function with lubridate equivalent (wday)
- §5.1 suggested adding both col names to by arg, instead of renaming cols
- §6 Added link to ap-tools-training page

Changes to `code_participant`:
- Updated code to reflect changes to Rmarkdown

Changes to `code_instructor()`:
- Added alternative solution to Ex §3.7.3
- §5.4.1 added alternative solution
- §5.4.3 added alternative solution
- Added `dplyr::` calls

Question:
- In §2.2 it is stated that "Note that if you are using R on the Analytical Platform the tidyverse package may already be installed, hence the above step can be skipped." But I thought that all environments are initially shipped without any non base R pacakges installed?

chloepugh commented on Intro_markdown.Rmd:
```diff
@@ -213,31 +209,31 @@ To know more about a package, it is always useful to read the associated documen
+Now look at the Console tab. You should see the commands library and `read_csv()` appear with the whole path to the data set. It is a good idea to copy and paste these commands inside your script, so you won't need to do this again to load the data.
 
-Alternatively you can simply use the function read_csv() from the readr package (which is included as part of tidyverse - see section 2.2):
+Alternatively you can simply use the function `read_csv()` from the `readr` package (which is included as part of tidyverse - see section 2.2:
```
Very minor, but looks like a bracket was removed from the end of this bit of text, and now means there's an open bracket with no matching close bracket.

chloepugh commented on Intro_markdown.Rmd:
````diff
@@ -213,31 +209,31 @@ To know more about a package, it is always useful to read the associated documen
 ```
 
-Both this and the code shown above are suitable for csv files, so it is assumed by default that the first line of the file contains a header (header = T) and the columns are separated by a comma symbol (sep = “,”). The latter approach assumes that you have already loaded the readr package, as it is part of tidyverse - see section 2.2.
+Both this and the code shown above are suitable for csv files, so it is assumed by default that the first line of the file contains a header (header = T) and the columns are separated by a comma symbol (sep = “,”). The former approach assumes that you have already loaded the `readr` package, as it is part of `tidyverse` - see section 2.2.
````
Re the final sentence - not sure if this is needed anymore, since that fact that readr is required for the former approach is already mentioned in the subsection above.

On the other hand, the sentence could be adapted to refer to the code in this section by changing `read.csv` to `readr::read_csv` (assuming that doesn't break anything).

Then it'd be more consistent if we use only `readr::read_csv` rather than introducing `read.csv` as well.

ms130 replied:
Good spot, have removed the redundant final sentence.

Good idea to make it consistent with `readr::read_csv`, but some columns get imported differently (no integer, numeric or character) so will have to update 2.5 as well (hopefully nothing else needs updating then too)

ms130 replied:
Think i've updated everything that needs to be

chloepugh reviewed (commented):
Changes look good! Just a couple of minor comments you might want to look at, then you should be good to merge :).

The other thing that might need updating is the README, as that still refers to the Word doc in a couple of places. It could be worth adding instructions on how to download the html material from github, in case it's not obvious (I have to right-click and select 'Save Link As...' before it'll let me download it normally in Firefox).

georginaeaton commented:
Sorry I've only got round to reviewing this now. It's probably a bit too late to merge this now as the training course is next Wednesday and most participants would have copied the repo across already. 

A couple of things that potentially need changing in the Read me and markdowns:
- When we talk about the prerequisite course - I think this is now called Introduction to using R on the AP with the repo link: https://github.com/moj-analytical-services/intro_using_r_on_ap
- Also, we need to take George Papadopoulos off as a contact - happy for this to be myself or you may wish to be the contact!

georginaeaton reviewed (changes requested):
A couple of things that potentially need changing in the Read me and markdowns:

When we talk about the prerequisite course - I think this is now called Introduction to using R on the AP with the repo link: https://github.com/moj-analytical-services/intro_using_r_on_ap

Also, we need to take George Papadopoulos off as a contact - happy for this to be myself or you may wish to be the contact!

With Q2 in section 4.2 exercises, can we say add a variable called **weekday** with the day of the week (as the solution adds a variable called weekday rather than day!)

chloepugh commented:
@ms130 @georginaeaton I've just added a commit to address Georgina's comments, so hopefully this is ready to merge now :).

I'm about to add some example code using Rs3tools, so will branch off this branch.
